src/App.tsx

```diff
@@ -1,18 +1,49 @@
-import React from 'react';
+import React, { useState } from 'react';
+import { useSearchParams } from 'react-router-dom';
 import './App.css';
 import { getNumbers } from './utils';
+import { Pagination } from './components/Pagination';
 
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
-const items = getNumbers(1, 42)
-  .map(n => `Item ${n}`);
+const items = getNumbers(1, 42);
 
 export const App: React.FC = () => {
+  const total = items.length;
+
+  const [searchParams, setSearchParams] = useSearchParams();
+
+  const [perPage, setPerPage] = useState(5);
+  const [currentPage, setCurrentPage] = useState(1);
+
+  const fromItem = (currentPage - 1) * perPage + 1;
+  const toItem = currentPage * perPage > total ? total : currentPage * perPage;
+
+  const visibleItems = getNumbers(fromItem, toItem);
+
+  const handleSelect = (e: React.ChangeEvent<HTMLSelectElement>) => {
+    setPerPage(Number(e.target.value));
+    setCurrentPage(1);
+
+    searchParams.set('page', '1');
+    searchParams.set('perPage', e.target.value);
+
+    setSearchParams(searchParams);
+  };
+
+  const handlePageChange = (page: number) => {
+    setCurrentPage(page);
+
+    searchParams.set('page', page.toString());
+    searchParams.set('perPage', perPage.toString());
+
+    setSearchParams(searchParams);
+  };
```

Comment:
You set query params but not apply them on page load. After reloading this params should set to state
![image](https://github.com/mate-academy/react_pagination/assets/54936044/91d96aad-6322-4575-ab54-a2728899527c)

Reply:
I'm not sure that I understood correctly what it should be, but set `page=2&perPage=7` in the URL and apply them on page load. 
A bit confusing why 2 and 7🤔 if first page=1 perPage=5

Reply:
`page=2&perPage=7` it's a example. It should work like this - when the page loads you take the query parameters from the url and write them into the appropriate state variables